---
title: "FINAL_REAL_log2_FC_Analysis_Promoter"
author: "Alexander Glahs"
date: "11 Mai 2019"
output:
  html_document:
    df_print: paged
  word_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r libraries,  warning=FALSE, message=FALSE}

#setwd("C:/Users/Admin/Documents/FINAL_REAL_PROMOTER_TISSUE_ANALYIS_WITH_OWN_RNASEQ/")

source("https://bioconductor.org/biocLite.R")
#biocLite("ggplot2")
#biocLite("heatmap.2")
#biocLite("heatmap")
#biocLite("fpc")
#biocLite("pvclust")
#biocLite("gplots")
#biocLite("rngtools")
#biocLite("kableExtra")

#library(ggplot2)

#library(heatmap)
#library(fpc)
library(pvclust)
#library(heatmap.2)
library(gplots)
#library(rngtools)
#library(NMF)

#biocLite("purrr")
#library(purrr)

#library(EDASeq)
library(fdrtool)
library(kableExtra)
library(knitr)
library(tidyr)
library(made4)
library(corrplot)
library(dplyr)
#library(tidyverse)
library(purrr)
library(dplyr)
#library(EDASeq)
library(fdrtool)
library(kableExtra)
library(knitr)
library(tidyr)
library(made4)
library(corrplot)
#library(reshape)
library(reshape2)

```

```{r LoadData,  warning=FALSE}



ID_table <-  read.table("C:/Users/Admin/Documents/Name_conversion_NM_FbGn_complete_unirot-gene_ID_master_list.csv",stringsAsFactors=F, sep= "\t",header=TRUE) 

data <- read.table("C:/Users/Admin/Documents/RNA_seq_Ali/File S1 - VST counts.csv",stringsAsFactors=F, sep= ",",header=TRUE) 
data_2 <- read.table("C:/Users/Admin/Documents/RNA_seq_Ali/File S1 - VST counts.csv",stringsAsFactors=F, sep= ",")

VST_counts <- data
colnames(VST_counts) <- data_2[1,]

#Load ChiP data
ChIP_data <- read.table("C:/Users/Admin/Documents/MultiBamCov_TSS_plus_200_NCBIrefseq_all_marks_all_TSS_pseudocount_dedup_dm6.txt"
                        ,stringsAsFactors=F, sep="\t",header=FALSE)
colnames(ChIP_data) <- c("Chr", "Start", "End","RefSeqID","Score","Strand","IC_4_6_K4me3_1","IC_4_6_K4me3_2","IC_6_8_K4me3_1","IC_6_8_K4me3_2","IC_8_10_K4me3_1","IC_8_10_K4me3_2","VC_4_6_K4me3_1","VC_4_6_K4me3_2","VC_68_K4me3_1_1","VC_68_K4me3_2_1","VC_810_K4me3_1_1","VC_810_K4me3_2_1","IC_4_6_K27ac_1","IC_4_6_K27ac_3","IC_6_8_K27ac_1","IC_6_8_K27ac_3","IC_8_10_K27ac_1","IC_8_10_K27ac_2","VC_4_6_K27ac_1","VC_4_6_K27ac_2","VC_68_K27ac_1_1","VC_68_K27ac_2_1","VC_810_K27ac_1_1","VC_810_K27ac_2_1","IC_4_6_K79me3_1","IC_4_6_K79me3_2","IC_6_8_K79me3_1","IC_6_8_K79me3_2","IC_8_10_K79me3_1","IC_8_10_K79me3_2","VC_4_6_K79me3_1","VC_4_6_K79me3_2","VC_68_K79me3_1_1","VC_68_K79me3_2_1","VC_810_K79me3_1_1","VC_810_K79me3_2_1","IC_4_6_K4mono_1","IC_4_6_K4mono_2","IC_6_8_K4mono_1","IC_6_8_K4mono_2","IC_8_10_K4mono_1","IC_8_10_K4mono_2","VC_4_6_K4mono_1","VC_4_6_K4mono_2","VC_68_K4mono_1_1","VC_68_K4mono_2_1","VC_810_K4mono_1_1","VC_810_K4mono_2_1","IC_4_6_K27me3_1","IC_4_6_K27me3_2","IC_6_8_K27me3_2","IC_6_8_K27me3_3","IC_8_10_K27me3_1","IC_8_10_K27me3_2","VC_4_6_K27me3_1","VC_4_6_K27me3_2","VC_68_K27me3_1_1","VC_68_K27me3_2_1","VC_810_K27me3_1_1","VC_810_K27me3_2_1"
)

#Load ID table
ID_table <-  read.table("C:/Users/Admin/Documents/Name_conversion_NM_FbGn_complete_unirot-gene_ID_master_list.csv",stringsAsFactors=F, sep= "\t",header=TRUE) 

l <- strsplit(as.character(ID_table$RefSeqID), ',')
ID_table_2 <- data.frame(Name = unlist(l), Name.string = rep(ID_table$RefSeqID, lengths(l)))
colnames(ID_table_2) <- c("RefSeq","RefSeqID")

ID_table_long <- merge(ID_table_2,ID_table,by=c("RefSeqID"),all=TRUE)

ID_table_long <- ID_table_long[!duplicated(ID_table_long[,c(2,4)]), ]
colnames(ID_table_long) <- c("All_RefSeq", "RefSeqID","Gene","FBgn")
ID_table_long$All_RefSeq <- paste0("'","'", ID_table_long$All_RefSeq, "'","'")
ID_table_long$Gene <- paste0("'","'", ID_table_long$Gene, "'","'")


colnames(ChIP_data) <- c("Chr", "Start", "End","RefSeqID","Score","Strand","IC_4_6_K4me3_1","IC_4_6_K4me3_2","IC_6_8_K4me3_1","IC_6_8_K4me3_2","IC_8_10_K4me3_1","IC_8_10_K4me3_2","VC_4_6_K4me3_1","VC_4_6_K4me3_2","VC_68_K4me3_1_1","VC_68_K4me3_2_1","VC_810_K4me3_1_1","VC_810_K4me3_2_1","IC_4_6_K27ac_1","IC_4_6_K27ac_3","IC_6_8_K27ac_1","IC_6_8_K27ac_3","IC_8_10_K27ac_1","IC_8_10_K27ac_2","VC_4_6_K27ac_1","VC_4_6_K27ac_2","VC_68_K27ac_1_1","VC_68_K27ac_2_1","VC_810_K27ac_1_1","VC_810_K27ac_2_1","IC_4_6_K79me3_1","IC_4_6_K79me3_2","IC_6_8_K79me3_1","IC_6_8_K79me3_2","IC_8_10_K79me3_1","IC_8_10_K79me3_2","VC_4_6_K79me3_1","VC_4_6_K79me3_2","VC_68_K79me3_1_1","VC_68_K79me3_2_1","VC_810_K79me3_1_1","VC_810_K79me3_2_1","IC_4_6_K4mono_1","IC_4_6_K4mono_2","IC_6_8_K4mono_1","IC_6_8_K4mono_2","IC_8_10_K4mono_1","IC_8_10_K4mono_2","VC_4_6_K4mono_1","VC_4_6_K4mono_2","VC_68_K4mono_1_1","VC_68_K4mono_2_1","VC_810_K4mono_1_1","VC_810_K4mono_2_1","IC_4_6_K27me3_1","IC_4_6_K27me3_2","IC_6_8_K27me3_2","IC_6_8_K27me3_3","IC_8_10_K27me3_1","IC_8_10_K27me3_2","VC_4_6_K27me3_1","VC_4_6_K27me3_2","VC_68_K27me3_1_1","VC_68_K27me3_2_1","VC_810_K27me3_1_1","VC_810_K27me3_2_1")

#Merge ChIP data with ID Table
chip_id_table <- merge(ChIP_data,ID_table_long,by= c("RefSeqID"),all.x=TRUE)

#Merge ChIP_ID data with RNA data
RNA_ChIP_data <- merge(chip_id_table,VST_counts,by=c("FBgn"),all.x=TRUE)

#RNA_ChIP_data_cor <- RNA_ChIP_data[!duplicated(RNA_ChIP_data$RefSeqID), ]




#Files from DeSeQ

EL_IC_K4me3 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_IC_K4me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_IC_K27ac <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_IC_K27ac_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_IC_K79me3 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_IC_K79me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_IC_K4me1 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_IC_K4me1_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_IC_K27me3 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_IC_K27me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 


EL_VC_K4me3 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_VC_K4me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_VC_K27ac <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_VC_K27ac_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_VC_K79me3 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_VC_K79me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_VC_K4me1 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_VC_K4me1_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

EL_VC_K27me3 <- read.table("C:/Users/Admin/Documents/FINAL_real_temporal_ANALYSIS_PROMOTER_DeSeq2/DESeq2_results_with_input_count_data_Temp_EL_VC_K27me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 



IC_46_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K4me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_46_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K4me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_68_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K4me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_68_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K4me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_810_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K4me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_810_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K4me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_46_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K27ac_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_46_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K27ac_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_68_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K27ac_IC_TOP100_basemean100_200bp_promotor_no_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_68_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K27ac_VC_TOP100_basemean100_200bp_promotor_no_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_810_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K27ac_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_810_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K27ac_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_46_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K79me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_46_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K79me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_68_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K79me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_68_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K79me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_810_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K79me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_810_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K79me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_46_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K4me1_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_46_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K4me1_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_68_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K4me1_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_68_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K4me1_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_810_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K4me1_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_810_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K4me1_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 


IC_46_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K27me3_IC_TOP100_basemean100_200bp_promotor_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_46_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_46h_K27me3_VC_TOP100_basemean100_200bp_promotor_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_68_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K27me3_IC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_68_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_68h_K27me3_VC_TOP100_basemean100_200bp_promotor.csv",stringsAsFactors=F, sep= "\t",header=T) 

IC_810_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K27me3_IC_TOP100_basemean100_200bp_promotor_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 
VC_810_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/Tissue_810h_K27me3_VC_TOP100_basemean100_200bp_promotor_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 


ID_column <- c(  rep("IC_46_K4me3",100),rep("VC_46_K4me3",100),rep("IC_68_K4me3",100),rep("VC_68_K4me3",100),rep("IC_810_K4me3",100),rep("VC_810_K4me3",100),rep("IC_46_K27ac",100),rep("VC_46_K27ac",100),rep("IC_68_K27ac",100),rep("VC_68_K27ac",100),rep("IC_810_K27ac",100),rep("VC_810_K27ac",100),rep("IC_46_K79me3",100),rep("VC_46_K79me3",100),rep("IC_68_K79me3",100),rep("VC_68_K79me3",100),rep("IC_810_K79me3",100),rep("VC_810_K79me3",100),rep("IC_46_K4me1",100),rep("VC_46_K4me1",100),rep("IC_68_K4me1",100),rep("VC_68_K4me1",100),rep("IC_810_K4me1",100),rep("VC_810_K4me1",100),rep("IC_46_K27me3",100),rep("VC_46_K27me3",100),rep("IC_68_K27me3",100),rep("VC_68_K27me3",100),rep("IC_810_K27me3",100),rep("VC_810_K27me3",100)  )


#Loading all promoters with a peak 

Tissue_46_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_46h_K4me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_68_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_68h_K4me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_810_K4me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_810h_K4me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 

Tissue_46_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_46h_K27ac_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_68_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_68h_K27ac_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_810_K27ac <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_810h_K27ac_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 

Tissue_46_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_46h_K79me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_68_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_68h_K79me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_810_K79me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_810h_K79me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 

Tissue_46_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_46h_K4me1_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_68_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_68h_K4me1_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_810_K4me1 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_810h_K4me1_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 

Tissue_46_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_46h_K27me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_68_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_68h_K27me3_promoters_200bp_with_RNA_seq_VST.csv",stringsAsFactors=F, sep= "\t",header=T) 
Tissue_810_K27me3 <- read.table("C:/Users/Admin/Documents/Final_Tissue_DeSeq_Tissue_Analysis/DESeq2_results_with_input_count_data_Tissue_810h_K27me3_promoters_200bp_with_RNA_seq_VST_NO_OUTLIER.csv",stringsAsFactors=F, sep= "\t",header=T) 





```

```{r CombineTOP, warning=FALSE}



All_top_100 <- rbind.data.frame(IC_46_K4me3,VC_46_K4me3,IC_68_K4me3,VC_68_K4me3,IC_810_K4me3,VC_810_K4me3,
                                IC_46_K27ac,VC_46_K27ac,IC_68_K27ac,VC_68_K27ac,IC_810_K27ac,VC_810_K27ac,
                                IC_46_K79me3,VC_46_K79me3,IC_68_K79me3,VC_68_K79me3,IC_810_K79me3,VC_810_K79me3,
                                IC_46_K4me1,VC_46_K4me1,IC_68_K4me1,VC_68_K4me1,IC_810_K4me1,VC_810_K4me1,
                                IC_46_K27me3,VC_46_K27me3,IC_68_K27me3,VC_68_K27me3,IC_810_K27me3,VC_810_K27me3)


TOP100_file_undup <- cbind(ID_column, All_top_100)

#Muss noch auf duplicate gefiltert werden

TOP100_file <- TOP100_file_undup[!duplicated(TOP100_file_undup$RefSeqID), ]


#List of genes and RefSeqs
unique__genes <- as.data.frame(unique(TOP100_file$Symbol))
#unique__genes
unique_Refseq <- as.data.frame(unique(TOP100_file$RefSeqID))
colnames(unique_Refseq) <- c("unique_Refseq_all")
#unique_Refseq


#Filter Tissue specific resuts
IC_all_upregulated_100 <- rbind.data.frame(IC_46_K4me3,IC_68_K4me3,IC_810_K4me3,
                                           IC_46_K27ac,IC_68_K27ac,IC_810_K27ac,
                                           IC_46_K79me3,IC_68_K79me3,IC_810_K79me3,
                                           VC_46_K4me1,VC_68_K4me1,VC_810_K4me1,
                                           VC_46_K27me3,VC_68_K27me3,VC_810_K27me3)
IC_all_upregulated_100 <- unique(IC_all_upregulated_100[,1:120])

VC_all_upregulated_100 <- rbind.data.frame(VC_46_K4me3,VC_68_K4me3,VC_810_K4me3,
                                           VC_46_K27ac,VC_68_K27ac,VC_810_K27ac,
                                           VC_46_K79me3,VC_68_K79me3,VC_810_K79me3,
                                           IC_46_K4me1,IC_68_K4me1,IC_810_K4me1,
                                           IC_46_K27me3,IC_68_K27me3,IC_810_K27me3)
VC_all_upregulated_100 <- unique(VC_all_upregulated_100[,1:120])

#VC_all_top_100 <- rbind.data.frame(VC_46_K4me3,VC_68_K4me3,VC_810_K4me3, VC_46_K27ac,VC_68_K27ac,VC_810_K27ac, VC_46_K79me3,VC_68_K79me3,VC_810_K79me3, VC_46_K4me1,VC_68_K4me1,VC_810_K4me1,VC_46_K27me3,VC_68_K27me3,VC_810_K27me3)

#Hey wenn ich alle signifikanten veränderungen in einen 3D vector space mappe kann ich Genactivität über die 
#Euklidische Distanz vorhersagen. Danke für die Inspiration.




VC_all_top_100 <- subset(TOP100_file_undup, log2FoldChange > 0)
IC_all_top_100 <- subset(TOP100_file_undup, log2FoldChange < 0)



#ID_column_VC = c("VC_46_K4me3","VC_68_K4me3","VC_810_K4me3"," VC_46_K27ac","VC_68_K27ac","VC_810_K27ac"," VC_46_K79me3","VC_68_K79me3","VC_810_K79me3","46_K4me1","68_K4me1","810_K4me1","46_K27me3","68_K27me3","810_K27me3") )


num_genes <- length(unique(TOP100_file$Symbol))
sig1 <- length(unique(TOP100_file$RefSeqID))
num_genes
sig1

ChiP_tables_MAT  <- data.matrix(TOP100_file[,9:68]) 

#ChiP_MAT <- na.omit(ChiP_tables_MAT) # listwise deletion of missing
#mydata <- scale(mydata[-1]) # standardize variables 

# Determine number of clusters
wss <- (nrow(ChiP_tables_MAT)-1)*sum(apply(ChiP_tables_MAT,2,var))
for (i in 1:20) wss[i] <- sum(kmeans(ChiP_tables_MAT,
                                     centers=i)$withinss)
plot(1:20, wss, type="b", xlab="Number of Clusters",
     ylab="Within groups sum of squares")


# K-Means Cluster Analysis
fit3 <- kmeans(TOP100_file[,9:68], 6) # 6 cluster solution
# get cluster means
aggregate(TOP100_file,by=list(fit3$cluster),FUN=mean)
# append cluster assignment
TOP100_file_6_kmeans <- data.frame(TOP100_file, fit3$cluster) 
#write.table(mydata, "C:/Users/aglahs/Downloads/TSS_plus_600_kmeans_clustering_7_clusters.txt", sep="\t") 

TOP100_file_6_kmeans <- TOP100_file_6_kmeans[order(TOP100_file_6_kmeans$fit3.cluster),]

heat_data_matrix <- data.matrix(log2(TOP100_file_6_kmeans[9:68]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs")



``` 


```{r ComboineALLfol2FC}

#List of all TSS with a peak


#Combined_large_list <- list(Tissue_46_K4me3[,c(1:7,118,119,122)],Tissue_68_K4me3[,c(1:7,118,119,122)],Tissue_810_K4me3[,c(1:7,118,119,122)],Tissue_46_K27ac[,c(1:7,118,119,122)],Tissue_68_K27ac[,c(1:7,118,119,122)],Tissue_810_K27ac[,c(1:7,118,119,122)],Tissue_46_K79me3[,c(1:7,118,119,122)],Tissue_68_K79me3[,c(1:7,118,119,122)],Tissue_810_K79me3[,c(1:7,118,119,122)],Tissue_46_K4me1[,c(1:7,118,119,122)],Tissue_68_K4me1[,c(1:7,118,119,122)],Tissue_810_K4me1[,c(1:7,118,119,122)],Tissue_46_K27me3[,c(1:7,118,119,122)],Tissue_68_K27me3[,c(1:7,118,119,122)],Tissue_810_K27me3[,c(1:7,118,119,122)]) %>% reduce(full_join, by = c("FBgn" , "RefSeqID" , "Chr" , "Start" , "End" , "Score" ,"Strand" ,"CGID" ,"Symbol"))                      

#colnames(Combined_large_list) = c("Fbgn", "RefSeqID","Chr","Start","End","Score","Strand","CGID","Symbol" ,"Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3")

RNA_ChIP_data_new  <- RNA_ChIP_data[order(RNA_ChIP_data$FBgn),]
RNA_ChIP_data_new  <- RNA_ChIP_data[,c(1:67,70:120)]
RNA_ChIP_data_new <- unique(RNA_ChIP_data_new)


#Merge_log2FC_data


TRIERBNF <- merge(RNA_ChIP_data,Tissue_46_K4me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_68_K4me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_810_K4me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_46_K27ac[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- unique(TRIERBNF) 
TRIERBNF <- merge(TRIERBNF,Tissue_68_K27ac[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_810_K27ac[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_46_K79me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_68_K79me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_810_K79me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- unique(TRIERBNF) 
TRIERBNF <- merge(TRIERBNF,Tissue_46_K4me1[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_68_K4me1[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_810_K4me1[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_46_K27me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- merge(TRIERBNF,Tissue_68_K27me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF <- unique(TRIERBNF) 
Combined_large_list <- merge(TRIERBNF,Tissue_810_K27me3[,c(2,122)], by= "RefSeqID", all.x =T)
Combined_large_list <- unique(Combined_large_list) 


colnames(Combined_large_list)[121:135] <- c("Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3")





#Replace all NA with 0
Combined_large_list[,121:135] <- replace(Combined_large_list[,121:135], is.na(Combined_large_list[,121:135]), 0)
#drops <- c("Symbol")
#Combined_large_list <- Combined_large_list[ , !(names(Combined_large_list) %in% drops)]






# K-Means Cluster Analysis
fit3 <- kmeans(Combined_large_list[,121:135], 6) # 6 cluster solution
# get cluster means
aggregate(Combined_large_list,by=list(fit3$cluster),FUN=mean)
# append cluster assignment
All_file_6_kmeans <- data.frame(Combined_large_list, fit3$cluster) 
#write.table(mydata, "C:/Users/aglahs/Downloads/TSS_plus_600_kmeans_clustering_7_clusters.txt", sep="\t") 

All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$fit3.cluster),]

#colnames(All_file_6_kmeans) = c("Fbgn","RefSeqID","Chr","Start","End","Score","Strand","CGID","Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3", "kmeans_6")

#Heatmap_kmenas

heat_data_matrix <- data.matrix(All_file_6_kmeans[121:135])
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-2, 2, length.out=101))


#colnames(All_file_6_kmeans) = c("Fbgn","RefSeqID","Chr","Start","End","Score","Strand","CGID","Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3", "kmeans_6")

#Heatmap custom sort

#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_810_K27ac),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_68_K27ac),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_46_K27ac),]

#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_46_K27me3),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_68_K27me3),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_810_K27me3),]

#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_810_K79me3),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_68_K79me3),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_46_K79me3),]

#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_810_K4me3),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_68_K4me3),]
#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$Tissue_46_K4me3),]

#heat_data_matrix <- data.matrix(All_file_6_kmeans[121:135])
#redgreen <- c("blue","white", "red") 
#pal <- colorRampPalette(redgreen)(100)
#heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-3, 3, length.out=101))

heat_data_matrix <- data.matrix(All_file_6_kmeans[8:67])
redgreen <- c("white","blue") 
pal <- colorRampPalette(redgreen)(200)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(0, 300, length.out=201))

heat_data_matrix <- data.matrix(All_file_6_kmeans[8:67])
redgreen <- c("white","blue") 
pal <- colorRampPalette(redgreen)(200)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(0, 500, length.out=201))

heat_data_matrix <- data.matrix(All_file_6_kmeans[8:67])
redgreen <- c("white","blue") 
pal <- colorRampPalette(redgreen)(200)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal,  ylab="TSSs",scale="column")


heat_data_matrix <- data.matrix(All_file_6_kmeans[8:67])
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(200)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal,  ylab="TSSs",scale="column")

heat_data_matrix <- data.matrix(All_file_6_kmeans[8:67])
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs")

heat_data_matrix <- data.matrix(All_file_6_kmeans[8:67])
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(1000)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs")

heat_data_matrix <- data.matrix(log2(All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(1000)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs")



fit3 <- kmeans(Combined_large_list[,121:135], 6) # 6 cluster solution
# get cluster means
aggregate(Combined_large_list,by=list(fit3$cluster),FUN=mean)
# append cluster assignment
All_file_6_kmeans <- data.frame(Combined_large_list, fit3$cluster) 
#write.table(mydata, "C:/Users/aglahs/Downloads/TSS_plus_600_kmeans_clustering_7_clusters.txt", sep="\t") 

All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$fit3),]

#colnames(All_file_6_kmeans) = c("Fbgn","RefSeqID","Chr","Start","End","Score","Strand","CGID","Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3", "kmeans_6")

#Heatmap_kmenas

heat_data_matrix <- data.matrix(All_file_6_kmeans[121:135])
redgreen <- c("blue","white", "red") 
col1 <- brewer.pal(6, "Set1")
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-2, 2, length.out=101),RowSideColors=col1[All_file_6_kmeans$fit3])

heat_data_matrix <- data.matrix(log2(All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit3])


#############

fit3 <- kmeans(Combined_large_list[,8:67], 8) # 6 cluster solution
# get cluster means
aggregate(Combined_large_list,by=list(fit3$cluster),FUN=mean)
# append cluster assignment
All_file_6_kmeans <- data.frame(Combined_large_list, fit3$cluster) 
#write.table(mydata, "C:/Users/aglahs/Downloads/TSS_plus_600_kmeans_clustering_7_clusters.txt", sep="\t") 

All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$fit3),]

#colnames(All_file_6_kmeans) = c("Fbgn","RefSeqID","Chr","Start","End","Score","Strand","CGID","Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3", "kmeans_6")

#Heatmap_kmenas

heat_data_matrix <- data.matrix(All_file_6_kmeans[121:135])
redgreen <- c("blue","white", "red") 
col1 <- brewer.pal(8, "Set1")
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-3, 3, length.out=101),RowSideColors=col1[All_file_6_kmeans$fit3])

heat_data_matrix <- data.matrix(log2(All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit3])

heat_data_matrix <- data.matrix((All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit3])


###########



fit3 <- kmeans(Combined_large_list[,121:135], 12) # 6 cluster solution
# get cluster means
aggregate(Combined_large_list,by=list(fit3$cluster),FUN=mean)
# append cluster assignment
All_file_6_kmeans <- data.frame(Combined_large_list, fit3$cluster) 
#write.table(mydata, "C:/Users/aglahs/Downloads/TSS_plus_600_kmeans_clustering_7_clusters.txt", sep="\t") 

All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$fit3),]

#colnames(All_file_6_kmeans) = c("Fbgn","RefSeqID","Chr","Start","End","Score","Strand","CGID","Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3", "kmeans_6")

#Heatmap_kmenas

heat_data_matrix <- data.matrix(All_file_6_kmeans[121:135])
redgreen <- c("blue","white", "red") 
col1 <- brewer.pal(12, "Set3")
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-3, 3, length.out=101),RowSideColors=col1[All_file_6_kmeans$fit3])

heat_data_matrix <- data.matrix(log2(All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit3])

heat_data_matrix <- data.matrix((All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit3])



#fit3 <- kmeans(Combined_large_list[,121:135], 12) # 6 cluster solution
#fit4 <- kmeans(Combined_large_list[,8:67], 12) # 6 cluster solution
# get cluster means
#aggregate(Combined_large_list,by=list(fit3$cluster),FUN=mean)

#aggregate(Combined_large_list,by=list(fit4$cluster),FUN=mean)
# append cluster assignment
#All_file_6_kmeans <- data.frame(Combined_large_list, fit4$cluster, fit3$cluster) 
#write.table(mydata, "C:/Users/aglahs/Downloads/TSS_plus_600_kmeans_clustering_7_clusters.txt", sep="\t") 

#All_file_6_kmeans <- All_file_6_kmeans[order(All_file_6_kmeans$fit4),]

#colnames(All_file_6_kmeans) = c("Fbgn","RefSeqID","Chr","Start","End","Score","Strand","CGID","Tissue_46_K4me3","Tissue_68_K4me3","Tissue_810_K4me3","Tissue_46_K27ac","Tissue_68_K27ac","Tissue_810_K27ac","Tissue_46_K79me3","Tissue_68_K79me3","Tissue_810_K79me3","Tissue_46_K4me1","Tissue_68_K4me1","Tissue_810_K4me1","Tissue_46_K27me3","Tissue_68_K27me3","Tissue_810_K27me3", "kmeans_6")

#Heatmap_kmenas

#heat_data_matrix <- data.matrix(All_file_6_kmeans[121:135])
#redgreen <- c("blue","white", "red") 
#col1 <- brewer.pal(12, "Set3")
#pal <- colorRampPalette(redgreen)(100)
#heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-3, 3, #length.out=101),RowSideColors=col1[All_file_6_kmeans$fit4])

#heat_data_matrix <- data.matrix(log2(All_file_6_kmeans[8:67]))
#redgreen <- c("white", "blue") 
#pal <- colorRampPalette(redgreen)(100)
#heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit4])

heat_data_matrix <- data.matrix((All_file_6_kmeans[8:67]))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs",RowSideColors=col1[All_file_6_kmeans$fit3])


```


```{r Euklidian distance}

#Calculate euklidian distance

#K27_vector = (All_file_6_kmeans$Tissue_46_K27ac + All_file_6_kmeans$Tissue_68_K27ac + All_file_6_kmeans$Tissue_810_K27ac) - (All_file_6_kmeans$Tissue_46_K27me3 + All_file_6_kmeans$Tissue_68_K27me3 + All_file_6_kmeans$Tissue_68_K27me3)

#K4_vector = (All_file_6_kmeans$Tissue_46_K4me3 + All_file_6_kmeans$Tissue_68_K4me3 + All_file_6_kmeans$Tissue_810_K4me3) - (All_file_6_kmeans$Tissue_46_K4me1 + All_file_6_kmeans$Tissue_68_K4me1 + All_file_6_kmeans$Tissue_810_K4me1)

#K79_vector =  (All_file_6_kmeans$Tissue_46_K79me3 + All_file_6_kmeans$Tissue_68_K79me3 + All_file_6_kmeans$Tissue_810_K79me3)

#Euklidian_dist = sqrt((K27_vector*K27_vector)+ (K4_vector*K4_vector) + (K79_vector*K79_vector))

#MAIN file
#Combine and sort file
#All_file_6_kmeans_euklid <- cbind(All_file_6_kmeans, K4_vector, K27_vector, K79_vector, Euklidian_dist)
#Order by euklidian dist
#All_file_6_kmeans_euklid <- All_file_6_kmeans_euklid[order(All_file_6_kmeans_euklid$Euklidian_dist),]


#Alternative_Euklidian_distance

#Combined_temporal_vectors
K4me3_vector = (All_file_6_kmeans$Tissue_46_K4me3 + All_file_6_kmeans$Tissue_68_K4me3 + All_file_6_kmeans$Tissue_810_K4me3)
K27ac_vector = (All_file_6_kmeans$Tissue_46_K27ac + All_file_6_kmeans$Tissue_68_K27ac + All_file_6_kmeans$Tissue_810_K27ac)
K79me3_vector = (All_file_6_kmeans$Tissue_46_K79me3 + All_file_6_kmeans$Tissue_68_K79me3 + All_file_6_kmeans$Tissue_810_K79me3)
K4me1_vector = (All_file_6_kmeans$Tissue_46_K4me1 + All_file_6_kmeans$Tissue_68_K4me1 + All_file_6_kmeans$Tissue_810_K4me1)
K27me3_vector = (All_file_6_kmeans$Tissue_46_K27me3 + All_file_6_kmeans$Tissue_68_K27me3 + All_file_6_kmeans$Tissue_810_K27me3)

Euklidian_dist = sqrt((K27ac_vector*K27ac_vector) +(K4me3_vector*K4me3_vector) + (K79me3_vector*K79me3_vector)+(K4me1_vector*K4me1_vector)+(K27me3_vector*K27me3_vector))

All_file_6_kmeans_euklid <- cbind(All_file_6_kmeans, K4me3_vector, K27ac_vector, K79me3_vector, K4me1_vector, K27me3_vector, Euklidian_dist)


write.table(All_file_6_kmeans_euklid,"Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid.csv", sep = "\t",quote=F, row.names=F)


#Individual vectors
K4em3_46_vector = All_file_6_kmeans$Tissue_46_K4me3
K27ac_46_vector = All_file_6_kmeans$Tissue_46_K27ac 
K79em3_46_vector = All_file_6_kmeans$Tissue_46_K79me3 
K4em1_46_vector = All_file_6_kmeans$Tissue_46_K4me1 
K27me3_46_vector = All_file_6_kmeans$Tissue_46_K27me3 
K4em3_68_vector = All_file_6_kmeans$Tissue_68_K4me3 
K27ac_68_vector = All_file_6_kmeans$Tissue_68_K27ac 
K79em3_68_vector = All_file_6_kmeans$Tissue_68_K79me3 
K4em1_68_vector = All_file_6_kmeans$Tissue_68_K4me1 
K27me3_68_vector = All_file_6_kmeans$Tissue_68_K27me3 
K4em3_810_vector = All_file_6_kmeans$Tissue_810_K4me3
K27ac_810_vector = All_file_6_kmeans$Tissue_810_K27ac
K79em3_810_vector = All_file_6_kmeans$Tissue_810_K79me3
K4em1_810_vector = All_file_6_kmeans$Tissue_810_K4me1
K27me3_810_vector = All_file_6_kmeans$Tissue_810_K27me3




#(Combine_wih_specific_TOP_100)?
Large_list_TOP_subset <- All_file_6_kmeans_euklid[All_file_6_kmeans_euklid$RefSeqID %in% as_vector(unique_Refseq), ] 


#Heatmap of euklidian sort

heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euklid[121:135])
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-5, 5, length.out=101))

uni_refSeq_df <- as.data.frame(unique_Refseq)
colnames(uni_refSeq_df) <- c("RefSeqID")
#Make summary dataframe and save to csv
Combined_TOP100_and_all_TSS <- left_join(uni_refSeq_df,All_file_6_kmeans_euklid, by =c( "RefSeqID" ))
#drops <- c("Chr.y","Start.y","End.y","Score.y","Strand.y","kmeans_6")
#Combined_TOP100_and_all_TSS <- Combined_TOP100_and_all_TSS[ , !(names(Combined_TOP100_and_all_TSS) %in% drops)]


Combined_TOP100_and_all_TSS <- Combined_TOP100_and_all_TSS[order(Combined_TOP100_and_all_TSS$Euklidian_dist),]

unique_Combined_TOP100_and_all_TSS <- unique(Combined_TOP100_and_all_TSS)

head(Combined_TOP100_and_all_TSS[,c(1:7,118:119)],n=20)

write.table(unique_Combined_TOP100_and_all_TSS,"Combined_TOP100_and_all_TSS_with_at_least_a_peak_with_all_Log2FC_summary_and_euclid.csv", sep = "\t",quote=F, row.names=F)


heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[121:135])
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-5, 5, length.out=101))

#test <- seq(-1,3, len=20)
#shift.BR <- colorRamp(c("blue","white", "red"), bias=0)((1:20)/20)
#tpal <- rgb(shift.BR, maxColorValue=255)
#barplot(test,col = tpal) 


heat_data_matrix_euklid <- data.matrix(Large_list_TOP_subset[121:135])
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid, Rowv=NA, Colv=NA, col=pal, scale="none",  ylab="TSSs", breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')


#Hirarchical clustering

heat_data_matrix_euklid <- data.matrix(Large_list_TOP_subset[121:135])
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid)))
heatmap(heat_data_matrix_euklid, Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

```

```{r Combine_all_with_euklid_with_OWN_DESeq_RNAseq}



Refseq_FBtr_table <- read.table("C:/Users/Admin/Documents/Ref_seq_and_FBtr_association_list_dm6_NM_NR.csv",stringsAsFactors=F, sep= "\t",header=TRUE)
colnames(Refseq_FBtr_table) <- c("RefSeqID","FBtr")
Combined_large_RNA_list_RNA <-  read.table("C:/Users/Admin/Documents/RNA_Seq_OWN_DESeq/Differential_RNA_seq_analysis_combined_DeSEq_list_FBgn.txt",stringsAsFactors=F, sep= "\t",header=TRUE)

MEGAMERCH <- merge(All_file_6_kmeans_euklid[,c(1:67,118:137:142)],Refseq_FBtr_table, by= "RefSeqID", all.x =T)
MEGAMERCH <- merge(MEGAMERCH,Combined_large_RNA_list_RNA, by= "FBtr", all.x =T)
MEGAMERCH <- unique(MEGAMERCH) 
MEGAMERCH <-  MEGAMERCH[order(-MEGAMERCH$Euklidian_dist),]

write.table(MEGAMERCH,"MASTER_Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_temp_sep_WITH_OWN_RNASEQ_DESEQ_FINAL.csv", sep = "\t",quote=F, row.names=F)

MEGAMERCH_ordered <- cbind(MEGAMERCH[,1:8],MEGAMERCH[,69:71],MEGAMERCH[,123:125],MEGAMERCH[,87:94],MEGAMERCH[,9:68],MEGAMERCH[,95:110],MEGAMERCH[,72:86], MEGAMERCH[,c(111,113,115,117,119,121)],MEGAMERCH[,c(112,114,116,118,120,122)])

write.table(MEGAMERCH_ordered,"MASTER_Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_temp_sep_WITH_OWN_RNASEQ_DESEQ_FINAL_ordered.csv", sep = "\t",quote=F, row.names=F)


```


```{r TOP100variablegenes}

unique_Combined_TOP100_and_all_TSS <- unique_Combined_TOP100_and_all_TSS[order(-unique_Combined_TOP100_and_all_TSS$Euklidian_dist),]


heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:100,121:135])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:100,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid)))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid, Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')



cat('\r\n\r\n')

#res_cor, 2)
par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:100,]$Symbol
heatmap.2(heat_data_matrix_euklid, cexRow=0.35, cexCol = 0.55,Colv=FALSE,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 100 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)

cat('\r\n\r\n')


par(oma=c(2,3,1,1))
#col.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:100,]$Symbol
heatmap.2((t(heat_data_matrix_euklid)), cexRow=0.65, cexCol = 0.45,Rowv=FALSE,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 100 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="col")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)

cat('\r\n\r\n')

#Top25
heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,121:135])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid)))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid, Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')


cat('\r\n\r\n')


#res_cor, 2)
par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
heatmap.2(heat_data_matrix_euklid[1:25,], cexRow=0.35, cexCol = 0.85,Colv=FALSE,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 25 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)

cat('\r\n\r\n')


par(oma=c(2,3,1,1))
#col.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:100,]$Symbol
heatmap.2((t(heat_data_matrix_euklid[1:25,])), cexRow=0.65, cexCol = 0.75,Rowv=FALSE,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 25 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="col")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)

cat('\r\n\r\n')




heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:125], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



#TOP40


heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:125], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')

#TOP50

heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:50,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:50,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:125], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:50,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:50,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:50,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:50,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


cat('\r\n\r\n')

redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:50,]$Symbol
heatmap.2(heat_data_matrix_euklid[1:50,111:125], cexRow=0.55, cexCol = 0.85,Colv=FALSE,Rowv = dd,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 50 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:50,]$Symbol
heatmap.2(heat_data_matrix_euklid[1:50,1:60], cexRow=0.55, cexCol = 0.85,Colv=FALSE,Rowv = dd,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 50 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)

par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:50,]$Symbol
heatmap.2(heat_data_matrix_euklid[1:50,61:109], cexRow=0.55, cexCol = 0.85,Colv=FALSE,Rowv = dd,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 50 differential TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal)#,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          #dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)

```

```{r Reorder_by_K27ac-K27me3}

All_file_6_kmeans_K27 <- cbind(All_file_6_kmeans_euklid, (abs(All_file_6_kmeans_euklid$K27ac_vector - All_file_6_kmeans_euklid$K27me3_vector)))

colnames(All_file_6_kmeans_K27)[143] = c("Combined_K27_vector")

All_file_6_kmeans_K27 <- All_file_6_kmeans_K27[order(-All_file_6_kmeans_K27$Combined_K27_vector),]


All_file_6_kmeans_K27[1:50,5]


#Top_results_on_K27vector

heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_K27[1:25,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_K27[1:25,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-8, 8, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')






#Top_results_on_K27vector

heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_K27[1:50,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_K27[1:50,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')

```

```{r individualTimepointsTOPs}

#Combined_seperated temporal_vectors



#Individual vectors
K4me3_46_vector = All_file_6_kmeans_K27$Tissue_46_K4me3
K27ac_46_vector = All_file_6_kmeans_K27$Tissue_46_K27ac 
K79me3_46_vector = All_file_6_kmeans_K27$Tissue_46_K79me3 
K4me1_46_vector = All_file_6_kmeans_K27$Tissue_46_K4me1 
K27me3_46_vector = All_file_6_kmeans_K27$Tissue_46_K27me3 
K4me3_68_vector = All_file_6_kmeans_K27$Tissue_68_K4me3 
K27ac_68_vector = All_file_6_kmeans_K27$Tissue_68_K27ac 
K79me3_68_vector = All_file_6_kmeans_K27$Tissue_68_K79me3 
K4me1_68_vector = All_file_6_kmeans_K27$Tissue_68_K4me1 
K27me3_68_vector = All_file_6_kmeans_K27$Tissue_68_K27me3 
K4me3_810_vector = All_file_6_kmeans_K27$Tissue_810_K4me3
K27ac_810_vector = All_file_6_kmeans_K27$Tissue_810_K27ac
K79me3_810_vector = All_file_6_kmeans_K27$Tissue_810_K79me3
K4me1_810_vector = All_file_6_kmeans_K27$Tissue_810_K4me1
K27me3_810_vector = All_file_6_kmeans_K27$Tissue_810_K27me3

Euklidian_dist_46 = sqrt((K27ac_46_vector*K27ac_46_vector) + (K79me3_46_vector*K79me3_46_vector)+(K4me1_46_vector*K4me1_46_vector)+(K27me3_46_vector*K27me3_46_vector))

Euklidian_dist_68 = sqrt((K27ac_68_vector*K27ac_68_vector) + (K79me3_68_vector*K79me3_68_vector)+(K4me1_68_vector*K4me1_68_vector)+(K27me3_68_vector*K27me3_68_vector))

Euklidian_dist_810 = sqrt((K27ac_810_vector*K27ac_810_vector) + (K79me3_810_vector*K79me3_810_vector)+(K4me1_810_vector*K4me1_810_vector)+(K27me3_810_vector*K27me3_810_vector))

All_file_6_kmeans_euk_K27_temp <- cbind(All_file_6_kmeans_K27, Euklidian_dist_46, Euklidian_dist_68, Euklidian_dist_810)





colnames(All_file_6_kmeans_euk_K27_temp)[144:146] = c("Euk46","Euk68","Euk810")



write.table(All_file_6_kmeans_euk_K27_temp,"Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_K27vector_temp_sep_FINAL.csv", sep = "\t",quote=F, row.names=F)


```

```{r Euklid46hours}

#Sorted by euklidian distance at 46h

All_file_6_kmeans_euk_K27_temp <- All_file_6_kmeans_euk_K27_temp[order(-All_file_6_kmeans_euk_K27_temp$Euk46),]



All_file_6_kmeans_euk_K27_temp[1:100,119]



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol
#Top_results_on 46 euklid
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,c(109,112,115,118,121)])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,c(109,112,115,118,121)], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



```

```{r Euklid68h}

#Sorted by euklidian distance at 68h



All_file_6_kmeans_euk_K27_temp <- All_file_6_kmeans_euk_K27_temp[order(-All_file_6_kmeans_euk_K27_temp$Euk68),]


All_file_6_kmeans_euk_K27_temp[1:100,119]


heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol
#Top_results_on 46 euklid
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,c(110,113,116,119,122)])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,c(110,113,116,119,122)], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')

All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol

```

```{r Eklid810h}

#Sorted by euklidian distance at 810h


All_file_6_kmeans_euk_K27_temp <- All_file_6_kmeans_euk_K27_temp[order(-All_file_6_kmeans_euk_K27_temp$Euk810),]


All_file_6_kmeans_euk_K27_temp[1:100,119]



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol
#Top_results_on 46 euklid
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,c(111,114,117,120,123)])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,c(111,114,117,120,123)], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')

All_file_6_kmeans_euk_K27_temp[1:40,]$Symbol

```

```{r TemporalseperatedVectors}

#Sorted by K27ac vector at 46h



All_file_6_kmeans_euk_K27_EUtemp_K27temp <- cbind(All_file_6_kmeans_euk_K27_temp, (abs(All_file_6_kmeans_euk_K27_temp$Tissue_46_K27ac - All_file_6_kmeans_euklid$Tissue_46_K27me3)))

colnames(All_file_6_kmeans_euk_K27_EUtemp_K27temp)[147] = c("K27_vector_46h")

All_file_6_kmeans_euk_K27_EUtemp_K27temp <- cbind(All_file_6_kmeans_euk_K27_EUtemp_K27temp, (abs(All_file_6_kmeans_euk_K27_EUtemp_K27temp$Tissue_68_K27ac - All_file_6_kmeans_euk_K27_EUtemp_K27temp$Tissue_68_K27me3)))

colnames(All_file_6_kmeans_euk_K27_EUtemp_K27temp)[148] = c("K27_vector_68h")

All_file_6_kmeans_euk_K27_EUtemp_K27temp <- cbind(All_file_6_kmeans_euk_K27_EUtemp_K27temp, (abs(All_file_6_kmeans_euk_K27_EUtemp_K27temp$Tissue_810_K27ac - All_file_6_kmeans_euk_K27_EUtemp_K27temp$Tissue_810_K27me3)))

colnames(All_file_6_kmeans_euk_K27_EUtemp_K27temp)[149] = c("K27_vector_810h")


#Sort by K27vector and plot

All_file_6_kmeans_euk_K27_EUtemp_K27temp <- All_file_6_kmeans_euk_K27_EUtemp_K27temp[order(-All_file_6_kmeans_euk_K27_EUtemp_K27temp$K27_vector_46h),]


All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:100,119]



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')

#Sort by 68h K27 vector
#Sorted by K27ac vector at 46h

All_file_6_kmeans_euk_K27_EUtemp_K27temp <- All_file_6_kmeans_euk_K27_EUtemp_K27temp[order(-All_file_6_kmeans_euk_K27_EUtemp_K27temp$K27_vector_68h),]


All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:100,5]



heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(log2(heat_data_matrix_euklid[,1:60]), Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')

#Sort by 810h K27 vector
#Sorted by K27ac vector at 810h

All_file_6_kmeans_euk_K27_EUtemp_K27temp <- All_file_6_kmeans_euk_K27_EUtemp_K27temp[order(-All_file_6_kmeans_euk_K27_EUtemp_K27temp$K27_vector_810h),]


All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:100,5]

write.table(All_file_6_kmeans_euk_K27_EUtemp_K27temp,"ENHANCER_Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_K27vector_temp_sep_K27vectortemp_Complete.csv", sep = "\t",quote=F, row.names=F)


heat_data_matrix_euklid <- data.matrix(All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:40,c(8:67,70:117,121:135)])
row.names(heat_data_matrix_euklid) <- All_file_6_kmeans_euk_K27_EUtemp_K27temp[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,109:123])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(log2(heat_data_matrix_euklid[,1:60]), Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')




```





```{r TemporalDateframes}

TRIERBNF2 <- merge(RNA_ChIP_data,EL_IC_K4me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_IC_K27ac[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_IC_K79me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_IC_K4me1[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- unique(TRIERBNF2) 
TRIERBNF2 <- merge(TRIERBNF2,EL_IC_K27me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_VC_K4me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_VC_K27ac[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_VC_K79me3[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_VC_K4me1[,c(2,122)], by= "RefSeqID", all.x =T)
TRIERBNF2 <- merge(TRIERBNF2,EL_VC_K27me3[,c(2,122)], by= "RefSeqID", all.x =T)
Combined_large_list2 <- unique(TRIERBNF2) 



colnames(Combined_large_list2)[121:130] <- c("EL_IC_K4me3","EL_IC_K27ac","EL_IC_K79me3","EL_IC_K4me1","EL_IC_K27me3","EL_VC_K4me3","EL_VC_K27ac","EL_VC_K79me3","EL_VC_K4me1","EL_VC_K27me3")



#Replace all NA with 0
Combined_large_list2[,121:130] <- replace(Combined_large_list2[,121:130], is.na(Combined_large_list2[,121:130]), 0)
#drops <- c("Symbol")
#Combined_large_list <- Combined_large_list[ , !(names(Combined_large_list) %in% drops)]






#Combined_temporal_vectors
T_K4me3_vector = (abs(Combined_large_list2$EL_IC_K4me3) + abs(Combined_large_list2$EL_VC_K4me3))
T_K27ac_vector = (abs(Combined_large_list2$EL_IC_K27ac) + abs(Combined_large_list2$EL_VC_K27ac))
T_K79me3_vector = (abs(Combined_large_list2$EL_IC_K79me3) + abs(Combined_large_list2$EL_VC_K79me3))
T_K4me1_vector = (abs(Combined_large_list2$EL_IC_K4me1) + abs(Combined_large_list2$EL_VC_K4me1))
T_K27me3_vector = (abs(Combined_large_list2$EL_IC_K27me3) + abs(Combined_large_list2$EL_VC_K27me3))

Euklidian_dist2 = sqrt((T_K27ac_vector*T_K27ac_vector) +(T_K4me3_vector*T_K4me3_vector) + (T_K79me3_vector*T_K79me3_vector)+(T_K4me1_vector*T_K4me1_vector)+(T_K27me3_vector*T_K27me3_vector))

Temp_All_file_6_euklid <- cbind(Combined_large_list2, T_K4me3_vector, T_K27ac_vector, T_K79me3_vector, T_K4me1_vector, T_K27me3_vector, Euklidian_dist2)


write.table(Temp_All_file_6_euklid,"Combined_Temporal_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid.csv", sep = "\t",quote=F, row.names=F)



Dist_IC_temp = (Combined_large_list2$EL_IC_K4me3 + Combined_large_list2$EL_IC_K27ac+ Combined_large_list2$EL_IC_K79me3 + Combined_large_list2$EL_IC_K4me1 - Combined_large_list2$EL_IC_K27me3)

Dist_VC_temp = (Combined_large_list2$EL_VC_K4me3 + Combined_large_list2$EL_VC_K27ac+ Combined_large_list2$EL_VC_K79me3 + Combined_large_list2$EL_VC_K4me1 - Combined_large_list2$EL_VC_K27me3)

Dist_sub_IC_VC <- Dist_IC_temp - Dist_VC_temp

Temp_All_file_6_euklid_dist <- cbind(Temp_All_file_6_euklid, Dist_IC_temp, Dist_VC_temp, Dist_sub_IC_VC)

Temp_All_file_6_euklid_dist <-  Temp_All_file_6_euklid_dist[order(-Temp_All_file_6_euklid_dist$Euklidian_dist2),]


####MAKE TEMPORAL MEGAMERCH WITH NEW RNA SEQ


TEMP_MEGAMERCH <- merge(Temp_All_file_6_euklid_dist[,c(1:67,118:137:139)],Refseq_FBtr_table, by= "RefSeqID", all.x =T)
TEMP_MEGAMERCH <- merge(TEMP_MEGAMERCH,Combined_large_RNA_list_RNA, by= "FBtr", all.x =T)
TEMP_MEGAMERCH <- unique(TEMP_MEGAMERCH) 
TEMP_MEGAMERCH <-  TEMP_MEGAMERCH[order(-TEMP_MEGAMERCH$Euklidian_dist2),]

write.table(TEMP_MEGAMERCH,"TEMP_MASTER_Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_temp_sep_WITH_OWN_RNASEQ_DESEQ_FINAL.csv", sep = "\t",quote=F, row.names=F)

TEMP_MEGAMERCH_ordered <- cbind(TEMP_MEGAMERCH[,1:8],TEMP_MEGAMERCH[,69:71],TEMP_MEGAMERCH[,120:122],TEMP_MEGAMERCH[,82:91],TEMP_MEGAMERCH[,9:68],TEMP_MEGAMERCH[,92:107],TEMP_MEGAMERCH[,72:81],TEMP_MEGAMERCH[,c(108,110,112,114,116,118)],TEMP_MEGAMERCH[,c(109,111,113,115,117,119)])

write.table(TEMP_MEGAMERCH_ordered,"MASTER_TEMP_Combined_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_temp_sep_WITH_OWN_RNASEQ_DESEQ_FINAL_ordered.csv", sep = "\t",quote=F, row.names=F)

MEGA_MEGAMERCH <- merge(MEGAMERCH_ordered,TEMP_MEGAMERCH_ordered[,c(1,2,15:23,101:110)], by= "RefSeqID", all.x =T)
MEGA_MEGAMERCH <- unique(MEGA_MEGAMERCH)
MEGA_MEGAMERCH <- subset(MEGA_MEGAMERCH, FBtr.x==FBtr.y)
MEGA_MEGAMERCH <- cbind(MEGA_MEGAMERCH[,c(1:22)],MEGA_MEGAMERCH[,c(126:135)],MEGA_MEGAMERCH[,c(23:125)],MEGA_MEGAMERCH[,c(136:145)])
names(MEGA_MEGAMERCH)[names(MEGA_MEGAMERCH) == "FBgn.x"] <- "FBgn"

write.table(MEGA_MEGAMERCH,"MASTER_MEGA_MEGAMERCH_no_states.csv", sep = "\t",quote=F, row.names=F)


write.table(Temp_All_file_6_euklid_dist,"Combined_Temporal_all_TSS_base_mean_100_with_all_Log2FC_summary_and_euclid_dist.csv", sep = "\t",quote=F, row.names=F)


GO_data <- read.delim("C:/Users/Admin/Documents/gene_onthology_per_gene.csv",stringsAsFactors=F, sep= "\t" ,header=TRUE) 
GO_descr <- read.delim("C:/Users/Admin/Documents/GO_description.csv",stringsAsFactors=F, sep= "\t" ,header=TRUE) 

names(GO_data)[names(GO_data) == "Fbgn"] <- "FBgn"

GO_MEGAMERCH <- merge(GO_data,GO_descr, by= "GO", all.x =T)
GO_MEGAMERCH <- merge(GO_MEGAMERCH,MEGA_MEGAMERCH, by= "FBgn", all.x =T)
GO_MEGAMERCH <- unique(GO_MEGAMERCH) 

write.table(GO_MEGAMERCH,"GO_MEGAMERCH.csv", sep = "\t",quote=F, row.names=F)

```

```{r MEGAMEGAMERCH_correlations}

#cor.plot(MEGA_MEGAMERCH[,109:123])


library(psych)

par(mar=c(0.1,0.1,0.1,0.1))
pairs.panels(MEGA_MEGAMERCH[,c(109:111,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)


pairs.panels(MEGA_MEGAMERCH[,c(112:114,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE)

pairs.panels(MEGA_MEGAMERCH[,c(115:117,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE)

pairs.panels(MEGA_MEGAMERCH[,c(118:120,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)


pairs.panels(MEGA_MEGAMERCH[,c(121:123,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE
             )


pairs.panels(MEGA_MEGAMERCH[,c(124:127,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE
             )


pairs.panels(MEGA_MEGAMERCH[,c(136:145)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE
             )




pairs.panels(MEGA_MEGAMERCH[,c(109,112,115,118,121,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)


pairs.panels(MEGA_MEGAMERCH[,c(110,113,116,119,122,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)

pairs.panels(MEGA_MEGAMERCH[,c(111,114,117,120,123,128,129)], 
             method = "pearson", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)




###Spearman instead pearson

par(mar=c(0.1,0.1,0.1,0.1))
pairs.panels(MEGA_MEGAMERCH[,c(109:111,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)


pairs.panels(MEGA_MEGAMERCH[,c(112:114,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE)

pairs.panels(MEGA_MEGAMERCH[,c(115:117,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE)

pairs.panels(MEGA_MEGAMERCH[,c(118:120,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)


pairs.panels(MEGA_MEGAMERCH[,c(121:123,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE
             )


pairs.panels(MEGA_MEGAMERCH[,c(124:127,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE
             )


pairs.panels(MEGA_MEGAMERCH[,c(136:145)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE, # show correlation ellipses
             stars=TRUE
             )




pairs.panels(MEGA_MEGAMERCH[,c(109,112,115,118,121,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)


pairs.panels(MEGA_MEGAMERCH[,c(110,113,116,119,122,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)

pairs.panels(MEGA_MEGAMERCH[,c(111,114,117,120,123,128,129)], 
             method = "spearman", # correlation method
             hist.col = "#00AFBB",
             density = TRUE,  # show density plots
             ellipses = TRUE , # show correlation ellipses
             stars=TRUE)



```




```{r Plot_Temp_data}

#Top temp genes with high variance
heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
redgreen <- c("blue", "white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')




#heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:75,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:75,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
#shift.BR <- colorRamp(c("blue","white", "red"), bias=1)((1:20)/20)
#tpal <- rgb(shift.BR, maxColorValue=255)
#heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=tpal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T)
#cat('\r\n\r\n')

#shift.BR <- colorRamp(c("white", "blue"), bias=1)((1:20)/20)
#tpal <- rgb(shift.BR, maxColorValue=255)
#heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=tpal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
#cat('\r\n\r\n')


#heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=tpal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


cat('\r\n\r\n')


redgreen <- c("blue", "white", "red") 
pal <- colorRampPalette(redgreen)(100)
par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[c(2,1,3:40),]$Symbol
heatmap.2(heat_data_matrix_euklid[c(2,1,3:40),111:120], cexRow=0.55, cexCol = 0.85,Colv=FALSE,Rowv = dd,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 50 temporal TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
heatmap.2(heat_data_matrix_euklid[1:40,1:60], cexRow=0.55, cexCol = 0.85,Colv=FALSE,Rowv = dd,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 50 temporal TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
#tracecol=NA, tracerow=NA)
cat('\r\n\r\n')

par(oma=c(2,1,1,1))
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
heatmap.2(heat_data_matrix_euklid[1:40,61:109], cexRow=0.55, cexCol = 0.85,Colv=FALSE,Rowv = dd,
          #cellnote = round(res_cor, 2),
         # notecex = 0.35, # same data set for cell labels
         main = "log2FC Top 50 temporal TSS", # heat map title
          #notecol="black",      # change font color of cell labels to black
         density.info="none",  # turns off density plot inside color legend
        trace="none",         # turns off trace lines inside the heat map
          #margins =c(12,9),     # widens margins around plot
          col=pal)#,       # use on color palette defined earlier
          #breaks=col_breaks,    # enable color transition at specified limits
          #dendrogram="row")    # only draw a row dendrogram
#Colv="NA")
cat('\r\n\r\n')




#Top temp genes with high variance between tissues
Temp_All_file_6_euklid_dist <-  Temp_All_file_6_euklid_dist[order(-abs(Temp_All_file_6_euklid_dist$Dist_sub_IC_VC)),]

heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
redgreen <- c("blue", "white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



#Top temp genes in IC late
Temp_All_file_6_euklid_dist <-  Temp_All_file_6_euklid_dist[order(-Temp_All_file_6_euklid_dist$Dist_IC_temp),]

heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



#Top temp genes in IC early
Temp_All_file_6_euklid_dist <-  Temp_All_file_6_euklid_dist[order(Temp_All_file_6_euklid_dist$Dist_IC_temp),]

heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')




#Top temp genes in VC late
Temp_All_file_6_euklid_dist <-  Temp_All_file_6_euklid_dist[order(-Temp_All_file_6_euklid_dist$Dist_VC_temp),]

heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



#Top temp genes in VC early
Temp_All_file_6_euklid_dist <-  Temp_All_file_6_euklid_dist[order(Temp_All_file_6_euklid_dist$Dist_VC_temp),]

heat_data_matrix_euklid <- data.matrix(Temp_All_file_6_euklid_dist[1:40,c(8:117,121:135)])
row.names(heat_data_matrix_euklid) <- Temp_All_file_6_euklid_dist[1:40,]$Symbol
dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:120])))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,111:120], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


heatmap(heat_data_matrix_euklid[,61:110], Colv=NA, col=pal, scale="row",  ylab="TSSs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')




```


```{r loadGranges}
#source("https://bioconductor.org/biocLite.R")
BiocManager::install("GenomicRanges")
library("GenomicRanges")
library("ChIPpeakAnno")

```


```{r readHMMtables}


#Load and grange object transformation

S10_50_master <- read.table("C:/Users/Admin/Documents/S10_50_master_file.bed",sep="\t",header=T,stringsAsFactors = FALSE)

gr_States <- makeGRangesFromDataFrame(S10_50_master, keep.extra.columns=TRUE)

AAA_Master <- cbind(All_file_6_kmeans_euk_K27_EUtemp_K27temp[,c(4,1:3,5:149)])

gr_promoters <- makeGRangesFromDataFrame(AAA_Master, keep.extra.columns=TRUE)





```

```{r Granges}


library(dplyr)

IC_46_promoter_state_regions <- filter(filter(S10_50_master, IC_46h <= 3),VC_46h ==5)
IC_68_promoter_state_regions <- filter(filter(S10_50_master, IC_68h <= 3),VC_68h ==5)
IC_810_promoter_state_regions <- filter(filter(S10_50_master, IC_810h <= 3),VC810h ==5)

VC_46_promoter_state_regions <- filter(filter(S10_50_master, VC_46h <= 3),IC_46h ==5)
VC_68_promoter_state_regions <- filter(filter(S10_50_master, VC_68h <= 3),IC_68h ==5)
VC_810_promoter_state_regions <- filter(filter(S10_50_master, VC810h <= 3),IC_810h ==5)

library(dplyr)

#All active genes for GO Analysis background



gr_IC_46_active_pr <- makeGRangesFromDataFrame(IC_46_promoter_state_regions, keep.extra.columns=TRUE)
countedoverl <- sum( countOverlaps(gr_promoters, gr_IC_46_active_pr))
print("IC_46:")
print(countedoverl)
#Subset hits and convert to dataframe
subset_IC_46_pr_active_str <- as(subsetByOverlaps(gr_promoters, gr_IC_46_active_pr),"data.frame")
write.table(subset_IC_46_pr_active_str, "IC_promoter_active_stringent_46_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

gr_IC_68_active_pr <- makeGRangesFromDataFrame(IC_68_promoter_state_regions, keep.extra.columns=TRUE)
countedoverl <- sum( countOverlaps(gr_promoters, gr_IC_68_active_pr))
print("IC_68:")
print(countedoverl)
#Subset hits and convert to dataframe
subset_IC_68_pr_active_str <- as(subsetByOverlaps(gr_promoters, gr_IC_68_active_pr),"data.frame")
write.table(subset_IC_68_pr_active_str, "IC_promoter_active_stringent_68_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

gr_IC_810_active_pr <- makeGRangesFromDataFrame(IC_810_promoter_state_regions, keep.extra.columns=TRUE)
countedoverl <- sum( countOverlaps(gr_promoters, gr_IC_810_active_pr))
print("IC_810:")
print(countedoverl)
#Subset hits and convert to dataframe
subset_IC_810_pr_active_str <- as(subsetByOverlaps(gr_promoters, gr_IC_810_active_pr),"data.frame")
write.table(subset_IC_810_pr_active_str, "IC_promoter_active_stringent_810_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)


gr_VC_46_active_pr <- makeGRangesFromDataFrame(VC_46_promoter_state_regions, keep.extra.columns=TRUE)
countedoverl <- sum( countOverlaps(gr_promoters, gr_VC_46_active_pr))
print("VC_46:")
print(countedoverl)
#Subset hits and convert to dataframe
subset_VC_46_pr_active_str <- as(subsetByOverlaps(gr_promoters, gr_VC_46_active_pr),"data.frame")
write.table(subset_VC_46_pr_active_str, "VC_promoter_active_stringent_46_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

gr_VC_68_active_pr <- makeGRangesFromDataFrame(VC_68_promoter_state_regions, keep.extra.columns=TRUE)
countedoverl <- sum( countOverlaps(gr_promoters, gr_VC_68_active_pr))
print("VC_68:")
print(countedoverl)
#Subset hits and convert to dataframe
subset_VC_68_pr_active_str <- as(subsetByOverlaps(gr_promoters, gr_VC_68_active_pr),"data.frame")
write.table(subset_VC_68_pr_active_str, "VC_promoter_active_stringent_68_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

gr_VC_810_active_pr <- makeGRangesFromDataFrame(VC_810_promoter_state_regions, keep.extra.columns=TRUE)
countedoverl <- sum( countOverlaps(gr_promoters, gr_VC_810_active_pr))
print("VC_810:")
print(countedoverl)
#Subset hits and convert to dataframe
subset_VC_810_pr_active_str <- as(subsetByOverlaps(gr_promoters, gr_VC_810_active_pr),"data.frame")
write.table(subset_VC_810_pr_active_str, "VC_promoter_active_stringent_810_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)





#Subset not overlaping genes from all active genes
#Tissue comparison
uniques_VC_46 <- setdiff(subsetByOverlaps(gr_promoters,gr_VC_46_active_pr),subsetByOverlaps(gr_promoters,gr_IC_46_active_pr))
uniques_out <- as(uniques_VC_46,"data.frame")
write.table(uniques_out, "Uniques_VC_vs_IC_46_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#uniques_out[,c(1:7,120)]

uniques_VC_68 <- setdiff(subsetByOverlaps(gr_promoters,gr_VC_68_active_pr),subsetByOverlaps(gr_promoters,gr_IC_68_active_pr))
uniques_out <- as(uniques_VC_68,"data.frame")
write.table(uniques_out, "Uniques_VC_vs_IC_68_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#uniques_out[,c(1:7,120)]

uniques_VC_810 <- setdiff(subsetByOverlaps(gr_promoters,gr_VC_810_active_pr),subsetByOverlaps(gr_promoters,gr_IC_810_active_pr))
uniques_out <- as(uniques_VC_810,"data.frame")
write.table(uniques_out, "Uniques_VC_vs_IC_810_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#uniques_out[,c(1:7,120)]

uniques_IC_46 <- setdiff(subsetByOverlaps(gr_promoters,gr_IC_46_active_pr),subsetByOverlaps(gr_promoters,gr_VC_46_active_pr))
uniques_out <- as(uniques_IC_46,"data.frame")
write.table(uniques_out, "Uniques_IC_vs_VC_46_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#uniques_out[,c(1:7,120)]

uniques_IC_68 <- setdiff(subsetByOverlaps(gr_promoters,gr_IC_68_active_pr),subsetByOverlaps(gr_promoters,gr_VC_68_active_pr))
uniques_out <- as(uniques_IC_68,"data.frame")
write.table(uniques_out, "Uniques_IC_vs_VC_68_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#uniques_out[,c(1:7,120)]

uniques_IC_810 <- setdiff(subsetByOverlaps(gr_promoters,gr_IC_810_active_pr),subsetByOverlaps(gr_promoters,gr_VC_810_active_pr))
uniques_out <- as(uniques_IC_810,"data.frame")
write.table(uniques_out, "Uniques_IC_vs_VC_810_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#uniques_out[,c(1:7,120)]

shared_uniques_IC <- subsetByOverlaps(uniques_IC_810, subsetByOverlaps(uniques_IC_46,uniques_IC_68))
shared_uniques_IC_df <- as(shared_uniques_IC,"data.frame")
write.table(shared_uniques_IC_df, "shared_Uniques_IC_betw_each_timep_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#shared_uniques_IC_df[,c(1:7,120)]

shared_uniques_VC <- subsetByOverlaps(uniques_VC_810, subsetByOverlaps(uniques_VC_46,uniques_VC_68))
shared_uniques_VC_df <- as(shared_uniques_VC,"data.frame")
write.table(shared_uniques_VC_df, "shared_Uniques_VC_betw_each_timep_active_prom_123to5.bed", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

#shared_uniques_VC_df[,c(1:7,120)]

#VENN DIAGRAMM

gr_act_pr_IC_46 <- unique(subsetByOverlaps(gr_promoters, gr_IC_46_active_pr))

gr_act_pr_IC_68 <- unique(subsetByOverlaps(gr_promoters, gr_IC_68_active_pr))

gr_act_pr_IC_810 <- unique(subsetByOverlaps(gr_promoters, gr_IC_810_active_pr))

gr_act_pr_VC_46 <- unique(subsetByOverlaps(gr_promoters, gr_VC_46_active_pr))

gr_act_pr_VC_68 <- unique(subsetByOverlaps(gr_promoters, gr_VC_68_active_pr))

gr_act_pr_VC_810 <- unique(subsetByOverlaps(gr_promoters, gr_VC_810_active_pr))



IC_res <- makeVennDiagram(Peaks=list(gr_act_pr_IC_46,gr_act_pr_IC_68, gr_act_pr_IC_810),
                          
                          NameOfPeaks=c("IC_46", "IC_68","IC_810"))

VC_res <- makeVennDiagram(Peaks=list(gr_act_pr_VC_46,gr_act_pr_VC_68, gr_act_pr_VC_810),
                          
                          NameOfPeaks=c("VC_46", "VC_68","VC_810"))


tissue_res <- makeVennDiagram(Peaks=list(gr_act_pr_VC_46,gr_act_pr_IC_46, gr_act_pr_VC_68,gr_act_pr_IC_68),
                              
                              NameOfPeaks=c("VC_46", "IC_46","VC_68", "IC_68"))



```

```{r IntegrateHMM}

library(bedr)


Intersect_master_states <- subsetByOverlaps(gr_States, gr_promoters)

Intersect_master <- mergeByOverlaps(gr_States, gr_promoters)

#IC46enh <- Intersect_master[ Intersect_master$IC_46h <= 3, ]
#IC46enh <- Intersect_master[ IC46enh$VC_46h == 5, ]

#IC68enh <- Intersect_master[ Intersect_master$IC_68h <= 3, ]
#IC68enh <- Intersect_master[ IC68enh$VC_68h == 5, ]

#IC810enh <- Intersect_master[ Intersect_master$IC_810h <= 3, ]
#IC810enh <- Intersect_master[ IC810enh$VC810h == 5, ]


#unique(IC46enh$gr_promoters$Symbol.x)

#unique(IC68enh$gr_promoters$Symbol.x)

#unique(IC810enh$gr_promoters$Symbol.x)


#VC46enh <- Intersect_master[ Intersect_master$VC_46h <= 3, ]
#IC46enh <- Intersect_master[ VC46enh$IC_46h == 5, ]

#VC68enh <- Intersect_master[ Intersect_master$VC_68h <= 3, ]
#VC68enh <- Intersect_master[ VC68enh$IC_68h == 5, ]

#VC810enh <- Intersect_master[ Intersect_master$VC810h <= 3, ]
#VC810enh <- Intersect_master[ VC810enh$IC_810h == 5, ]


#unique(VC46enh$gr_promoters$Symbol.x)

#unique(VC68enh$gr_promoters$Symbol.x)

#unique(VC810enh$gr_promoters$Symbol.x)

#gr_IC46enh <-  unique(IC46enh$gr_promoters)

#gr_IC68enh <-  unique(IC68enh$gr_promoters)

#gr_IC810enh <-  unique(IC810enh$gr_promoters)


#gr_VC46enh <-  unique(VC46enh$gr_promoters)

#gr_VC68enh <-  unique(VC68enh$gr_promoters)

#gr_VC810enh <-  unique(VC810enh$gr_promoters)


Venn_enhancer_VC <- makeVennDiagram(Peaks=list(gr_act_pr_VC_46,gr_act_pr_VC_68, gr_act_pr_VC_810),
                          
                          NameOfPeaks=c("VC_46", "VC_68","VC_810"))


Venn_enhancer_VC <- makeVennDiagram(Peaks=list(gr_act_pr_IC_46,gr_act_pr_IC_68, gr_act_pr_IC_810),
                          
                          NameOfPeaks=c("IC_46", "IC_68","IC_810"))

#Shared hit genes Check out the cadherin complex


shared_uniques_VC <- subsetByOverlaps(gr_act_pr_VC_46, subsetByOverlaps(gr_act_pr_VC_68,gr_act_pr_VC_810))
shared_uniques_VC_df <- as(shared_uniques_VC,"data.frame")
shared_uniques_VC_df[,c(1:7,120)]
shared_uniques_VC610 <- subsetByOverlaps(gr_act_pr_VC_68,gr_act_pr_VC_810)
shared_uniques_VC_6to10 <- as(shared_uniques_VC610,"data.frame")
shared_uniques_VC_6to10[,c(1:7,120)]
shared_uniques_VC48 <- subsetByOverlaps(gr_act_pr_VC_46,gr_act_pr_VC_68)
shared_uniques_VC_4to8 <- as(shared_uniques_VC48,"data.frame")
shared_uniques_VC_4to8[,c(1:7,120)]
shared_uniques_VC410 <- subsetByOverlaps(gr_act_pr_VC_46,gr_act_pr_VC_810)
shared_uniques_VC_4to10 <- as(shared_uniques_VC410,"data.frame")
shared_uniques_VC_4to10[,c(1:7,120)]


heat_data_matrix_euklid <- data.matrix(as.data.frame(shared_uniques_VC_df[,c(8:67,70:117,121:135)]))
row.names(heat_data_matrix_euklid) <- shared_uniques_VC_df$Symbol
dd <- as.dendrogram(hclust(dist(as.data.frame(shared_uniques_VC_df[,121:135]))))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="CRMs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="CRMs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



shared_uniques_IC <- subsetByOverlaps(gr_act_pr_IC_46, subsetByOverlaps(gr_act_pr_IC_68,gr_act_pr_IC_810))
shared_uniques_IC_df <- as(shared_uniques_IC,"data.frame")
shared_uniques_IC_df[,c(1:7,120)]
shared_uniques_IC610 <- subsetByOverlaps(gr_act_pr_IC_68,gr_act_pr_IC_810)
shared_uniques_IC_6to10 <- as(shared_uniques_IC610,"data.frame")
shared_uniques_IC_6to10[,c(1:7,120)]
shared_uniques_IC48 <- subsetByOverlaps(gr_act_pr_IC_46,gr_act_pr_IC_68)
shared_uniques_IC_4to8 <- as(shared_uniques_IC48,"data.frame")
shared_uniques_IC_4to8[,c(1:7,120)]
shared_uniques_IC410 <- subsetByOverlaps(gr_act_pr_IC_46,gr_act_pr_IC_810)
shared_uniques_IC_4to10 <- as(shared_uniques_IC410,"data.frame")
shared_uniques_IC_4to10[,c(1:7,120)]





heat_data_matrix_euklid <- data.matrix(as.data.frame(shared_uniques_IC_df[,c(8:67,70:117,121:135)]))
row.names(heat_data_matrix_euklid) <- shared_uniques_IC_df$Symbol
dd <- as.dendrogram(hclust(dist(as.data.frame(shared_uniques_IC_df[,121:135]))))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSS",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="TSS",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="TSS",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


par( mfrow = c( 1, 4 ) )
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_IC_df[,73]+shared_uniques_IC_df[,74])/2)/((shared_uniques_IC_df[,81]+shared_uniques_IC_df[,82])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")

par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_IC_df[,77]+shared_uniques_IC_df[,78])/2)/((shared_uniques_IC_df[,85]+shared_uniques_IC_df[,86])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")


par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_VC_df[,73]+shared_uniques_VC_df[,74])/2)/((shared_uniques_VC_df[,81]+shared_uniques_VC_df[,82])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")

par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_VC_df[,77]+shared_uniques_VC_df[,78])/2)/((shared_uniques_VC_df[,85]+shared_uniques_VC_df[,86])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")




#IC specific over neg
par( mfrow = c( 1, 2 ) )
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_IC_df[,73]+shared_uniques_IC_df[,74])/2)/((shared_uniques_IC_df[,71]+shared_uniques_IC_df[,72])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")

#VC specific over neg
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_VC_df[,81]+shared_uniques_VC_df[,82])/2)/((shared_uniques_VC_df[,79]+shared_uniques_VC_df[,80])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")



```

```{r InnerGenes2hitsHMM}

#All genes found in at least 2 timepoints
Inner_join_IC <- unique(rbind(shared_uniques_IC_6to10,shared_uniques_IC_4to10,shared_uniques_IC_4to8))


Inner_join_VC <- unique(rbind(shared_uniques_VC_6to10,shared_uniques_VC_4to10,shared_uniques_VC_4to8))


heat_data_matrix_euklid <- data.matrix(as.data.frame(Inner_join_IC[,c(8:67,70:117,121:135)]))
row.names(heat_data_matrix_euklid) <- Inner_join_IC$Symbol
dd <- as.dendrogram(hclust(dist(as.data.frame(Inner_join_IC[,121:135]))))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="CRMs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="CRMs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')



heat_data_matrix_euklid <- data.matrix(as.data.frame(Inner_join_VC[,c(8:67,70:117,121:135)]))
row.names(heat_data_matrix_euklid) <- Inner_join_VC$Symbol
dd <- as.dendrogram(hclust(dist(as.data.frame(Inner_join_VC[,121:135]))))
redgreen <- c("blue","white", "red") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,109:123], Colv=NA, col=pal, scale="none",  ylab="TSSs",Rowv=dd,keep.dendro=T, breaks=seq(-5, 5, length.out=101))
cat('\r\n\r\n')

#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,1:60], Colv=NA, col=pal, scale="none",  ylab="CRMs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')


#heat_data_matrix_euklid <- data.matrix(unique_Combined_TOP100_and_all_TSS[1:25,c(8:117,121:135)])
#row.names(heat_data_matrix_euklid) <- unique_Combined_TOP100_and_all_TSS[1:25,]$Symbol
#dd <- as.dendrogram(hclust(dist(heat_data_matrix_euklid[,111:125])))
redgreen <- c("white", "blue") 
pal <- colorRampPalette(redgreen)(100)
heatmap(heat_data_matrix_euklid[,61:108], Colv=NA, col=pal, scale="row",  ylab="CRMs",Rowv=dd,keep.dendro=T)
cat('\r\n\r\n')




par( mfrow = c( 1, 4 ) )
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_IC_df[,73]+shared_uniques_IC_df[,74])/2)/((shared_uniques_IC_df[,81]+shared_uniques_IC_df[,82])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")

par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_IC_df[,77]+shared_uniques_IC_df[,78])/2)/((shared_uniques_IC_df[,85]+shared_uniques_IC_df[,86])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")


par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_VC_df[,73]+shared_uniques_VC_df[,74])/2)/((shared_uniques_VC_df[,81]+shared_uniques_VC_df[,82])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")

par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_VC_df[,77]+shared_uniques_VC_df[,78])/2)/((shared_uniques_VC_df[,85]+shared_uniques_VC_df[,86])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")




#IC specific over neg
par( mfrow = c( 1, 2 ) )
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_IC_df[,73]+shared_uniques_IC_df[,74])/2)/((shared_uniques_IC_df[,71]+shared_uniques_IC_df[,72])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")

#VC specific over neg
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot(log2(((shared_uniques_VC_df[,81]+shared_uniques_VC_df[,82])/2)/((shared_uniques_VC_df[,79]+shared_uniques_VC_df[,80])/2)),ylim=c(-0.4,0.4))
abline(h = 0, col = "red")


#IC in at least 2
Inner_join_IC$Symbol
cat('\r\n\r\n')

#VC in at least 2
Inner_join_VC$Symbol
cat('\r\n\r\n')

#IC in at least 2
Inner_join_IC$RefSeqID
cat('\r\n\r\n')

#VC in at least 2
Inner_join_VC$RefSeqID
cat('\r\n\r\n')

#Outer_join
#IC
unique(c(gr_act_pr_IC_46$Symbol,gr_act_pr_IC_68$Symbol,gr_act_pr_IC_810$Symbol))
cat('\r\n\r\n')
#VC
unique(c(gr_act_pr_VC_46$Symbol,gr_act_pr_VC_68$Symbol,gr_act_pr_VC_810$Symbol))

```

```{r Log2FC_forshared_unique_df}


#All_hmm_hits_extracted

IC_HMM_hits <- unique(rbind(as.data.frame(gr_act_pr_IC_46), as.data.frame(gr_act_pr_IC_68),as.data.frame(gr_act_pr_IC_810) ))
VC_HMM_hits <- unique(rbind(as.data.frame(gr_act_pr_VC_46), as.data.frame(gr_act_pr_VC_68),as.data.frame(gr_act_pr_VC_810) ))





IC_46_HMM_hits <- unique(as.data.frame(gr_act_pr_IC_46))
IC_46_HMM_hits <-(unique(IC_46_HMM_hits[,c(1:7)]))
IC_68_HMM_hits <- unique(as.data.frame(gr_act_pr_IC_68))
IC_68_HMM_hits <-(unique(IC_68_HMM_hits[,c(1:7)]))
IC_810_HMM_hits <- unique(as.data.frame(gr_act_pr_IC_810))
IC_810_HMM_hits <-(unique(IC_810_HMM_hits[,c(1:7)]))

VC_46_HMM_hits <- unique(as.data.frame(gr_act_pr_VC_46))
VC_46_HMM_hits <-(unique(VC_46_HMM_hits[,c(1:7)]))
VC_68_HMM_hits <- unique(as.data.frame(gr_act_pr_VC_68))
VC_68_HMM_hits <-(unique(VC_68_HMM_hits[,c(1:7)]))
VC_810_HMM_hits <- unique(as.data.frame(gr_act_pr_VC_810))
VC_810_HMM_hits <-(unique(VC_810_HMM_hits[,c(1:7)]))

write.table(IC_46_HMM_hits, "C:/Users/Admin/Downloads/IC_46_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

write.table(IC_68_HMM_hits, "C:/Users/Admin/Downloads/IC_68_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

write.table(IC_810_HMM_hits, "C:/Users/Admin/Downloads/IC_810_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

write.table(VC_46_HMM_hits, "C:/Users/Admin/Downloads/VC_46_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

write.table(VC_68_HMM_hits, "C:/Users/Admin/Downloads/VC_68_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

write.table(VC_810_HMM_hits, "C:/Users/Admin/Downloads/VC_810_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)


IC_HMM_MEGA_MEGAMERCH <- merge(IC_HMM_hits[,c(1:7)],MEGA_MEGAMERCH, by= "RefSeqID", all.x =F)

VC_HMM_MEGA_MEGAMERCH <- merge(VC_HMM_hits[,c(1:7)],MEGA_MEGAMERCH, by= "RefSeqID", all.x =F)


IC_HMM_MEGA_MEGAMERCH[, 39:114] <- log(IC_HMM_MEGA_MEGAMERCH[, 39:114], 2)



write.table(IC_HMM_MEGA_MEGAMERCH, "C:/Users/Admin/Downloads/IC_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)
write.table(VC_HMM_MEGA_MEGAMERCH, "C:/Users/Admin/Downloads/VC_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)


IC_HMM_MEGA_MEGAMERCH[, 39:114] <- log(IC_HMM_MEGA_MEGAMERCH[, 39:114], 2)
VC_HMM_MEGA_MEGAMERCH[, 39:114] <- log(VC_HMM_MEGA_MEGAMERCH[, 39:114], 2)


write.table(IC_HMM_MEGA_MEGAMERCH, "C:/Users/Admin/Downloads/IC_HMM_hits_Master_list_Promoters_raw_log2.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)
write.table(VC_HMM_MEGA_MEGAMERCH, "C:/Users/Admin/Downloads/VC_HMM_hits_Master_list_Promoters_raw_log2.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)


VC_Inner_join_HMM_MEGA_MEGAMERCH <- merge(Inner_join_VC[,c(1:7)],MEGA_MEGAMERCH, by= "RefSeqID", all.x =F)

IC_Inner_join_HMM_MEGA_MEGAMERCH <- merge(Inner_join_IC[,c(1:7)],MEGA_MEGAMERCH, by= "RefSeqID", all.x =F)

write.table(IC_Inner_join_HMM_MEGA_MEGAMERCH, "C:/Users/Admin/Downloads/IC_Inner_join_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)
write.table(VC_Inner_join_HMM_MEGA_MEGAMERCH, "C:/Users/Admin/Downloads/VC_Inner_join_HMM_hits_Master_list_Promoters.txt", col.names=TRUE, sep="\t",row.names=FALSE,quote=FALSE)

```

```{r NEW_FIGURESDESEQ}

#IC specific over neg
par( mfrow = c( 1, 4 ) )
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot((IC_Inner_join_HMM_MEGA_MEGAMERCH[,130:135]*-1),ylim=c(-3,7))
abline(h = 0, col = "red")

#VC specific over neg
#par( mfrow = c( 1, 2 ) )
#par(las=2)
#par(mar=c(10, 4, 0.5, 1))
boxplot((VC_Inner_join_HMM_MEGA_MEGAMERCH[,130:135]*-1),ylim=c(-5,7))
abline(h = 0, col = "red")



#ALL IC specific over neg
#par( mfrow = c( 1, 2 ) )
par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot((IC_HMM_MEGA_MEGAMERCH[,130:135]*-1),ylim=c(-7,7))
abline(h = 0, col = "red")

#ALL VC specific over neg
#par( mfrow = c( 1, 2 ) )
#par(las=2)
par(mar=c(10, 4, 0.5, 1))
boxplot((VC_HMM_MEGA_MEGAMERCH[,130:135]*-1),ylim=c(-7,7))
abline(h = 0, col = "red")


nrow(IC_Inner_join_HMM_MEGA_MEGAMERCH)

nrow(VC_Inner_join_HMM_MEGA_MEGAMERCH)

nrow(IC_HMM_MEGA_MEGAMERCH)

nrow(VC_HMM_MEGA_MEGAMERCH)




```



